File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        ret = 0
        l,r = 0,0
        n = len(nums)
        flips = 0
        while r<n:
            if nums[r] == 0:
                flips +=1
            while flips>k:
                if nums[l] == 0:
                    flips-=1
                l+=1
            ret = max(ret,r-l+1)
            r+=1
        return ret

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.ts
function longestOnes(nums: number[], k: number): number {
    let ret = 0
    let l =0
    let r = 0
    let flips = 0
    while (r<nums.length){
        if (nums[r] != 1){
            flips++
        }
        while(flips>k){
            if (nums[l] !=1){
                flips--
            }
            l++
        }
        ret = Math.max(r-l+1,ret)
        r++
    }
    return ret
};